### PR TITLE
Experimental support for 5x5 convolution on CPU

### LINKED ISSUE #475: support 5x5 convolution
In #435, I made a new network which uses 5x5 convolution. Recently I did some experiments to remove 5x5 conv from the network, but results were not good.

Therefore, I want support of 5x5 convolution. (stride=1 would be fine, of course happier with stride=2 support.)

### PR DESCRIPTION
**Pull request by primenumber**
part of #475 .

## Motivation and Context

5x5 convolution is desired by #435 and other networks.
Almost of all routines are already supports 5x5 convolution, but not tested well...

## Description

In 5x5 convolution, maximum number of channel is limited by 436 (= floor(32767 / (5*5 * 3))).
So assertion are used to check this constraints.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature / Optimization (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by primenumber** [comment]
How to test 5x5 conv...?
Do you have any idea? @tkng

**Comment by primenumber** [comment]
notice: current FPGA implementation cannot execute 5x5 conv

File: dlk/python/dlk/templates/include/de10_nano.h [hunk 1]
```diff
@@ -146,9 +146,14 @@ Parameters calcParameters(uint32_t inputHeight, uint32_t inputWidth, uint32_t in
   constexpr uint32_t maxBurst = 32;
   constexpr uint32_t b = 32;
 
-  assert((kernelHeight == 3 && kernelWidth == 3) || (kernelHeight == 1 && kernelWidth == 1));
-
-  uint32_t pad = (kernelHeight == 1) ? 0 : 1;
+  constexpr std::size_t n_bit = 2;
+  constexpr std::size_t maxA = (1 << n_bit) - 1;
+  assert(kernelHeight == kernelWidth); // kernel rectangle must be square
+  assert(kernelHeight % 2 == 1); // kernel size must be even
```
Comment (joelN123) [inline dlk/python/dlk/templates/include/de10_nano.h]: the comment should be 'must be odd' I think?

Comment (primenumber) [reply dlk/python/dlk/templates/include/de10_nano.h]: Oh! yes!

Comment (joelN123) [reply dlk/python/dlk/templates/include/de10_nano.h]: ahh, I should not be reviewer, I just happened to see by chance.

Comment (primenumber) [reply dlk/python/dlk/templates/include/de10_nano.h]: Fixed.

**Comment by primenumber** [comment]
I gave up 5x5 conv on TCAv2 by this branch.
5x5 conv for TCAv2 may be supported later PR.

**Pull request merged**